src/main/kotlin/org/rust/lang/core/psi/RustNumber.kt

```diff
@@ -0,0 +1,79 @@
+package org.rust.lang.core.psi
+
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.RustNumber.Float.F32
+import org.rust.lang.core.psi.RustNumber.Float.F64
+import org.rust.lang.core.psi.RustNumber.Int.*
+import org.rust.lang.core.psi.RustTokenElementTypes.FLOAT_LITERAL
+import org.rust.lang.core.psi.RustTokenElementTypes.INTEGER_LITERAL
+import java.lang.Long.parseLong
+
+sealed class RustNumber<out T>(val valueString: String) {
```

Comment:
Hm, it's still looks a bit heavy. This generics don't really by any type safety because we use them with `<*>` and do a cast to `Double` anyway. Moreover, there should be only `RustNumber<Long>` and `RustNumber<Double>`, that is, the set of possible values for `T` is closed. Can we remove this class altogether, and have

````Kotlin
fun valueAsDouble(): Doube?
fun valueAsLong(): Long?
```

on `RustNumericLiteral` directly?